### PR TITLE
Add support for several Windows types

### PR DESCRIPTION
**Pull request by dcristoloveanu**
Add support for several Windows types

### CODE REVIEW

File: inc/umock_c/umock_c.h [hunk 1]
```diff
@@ -130,11 +130,14 @@ int umock_c_set_call_recorder(UMOCKCALLRECORDER_HANDLE umockc_call_recorder);
     MU_IF(MU_DIV2(MU_COUNT_ARG(__VA_ARGS__)), REGISTER_UMOCK_VALUE_TYPE_ALL, REGISTER_UMOCK_VALUE_TYPE_ONLY_TYPE) (__VA_ARGS__)
 #endif
 
+#define UMOCK_INTERNAL_WINDOWS_REGISTER_ALIAS_TYPE(registered_type, is_type) \
```
Comment (mattdurak) [inline inc/umock_c/umock_c.h]: >_WINDOWS [](start = 22, length = 8)

doesn't look Windows specific here

File: devdoc/umock_c_lib_requirements_with_ids.md [hunk 2]
```diff
@@ -444,6 +444,15 @@ XX**SRS_UMOCK_C_LIB_01_148: [** If call comparison fails an error shall be indic
...
 **SRS_UMOCK_C_LIB_01_153: [** If no custom handler has beed registered for a pointer type, it shall be trated as void*. **]**
 
+### Windows types
+
+**SRS_UMOCKTYPES_WINDOWS_01_001: [** The following Windows types shall be supported out of the box, aliased to their underlying types: **]**
+
+- LONG (long)
+- LONGLONG (long long)
+- DWORD (unsigned long)
+- HRESULT (LONG)
```
Comment (mattdurak) [inline devdoc/umock_c_lib_requirements_with_ids.md]: >HRESULT [](start = 2, length = 7)

Doesn't seem to cover all of the types that are actually supported

Comment (dcristoloveanu) [reply devdoc/umock_c_lib_requirements_with_ids.md]: Added all of them.

File: inc/umock_c/umock_c.h (hunk 1 above)
Comment (dcristoloveanu) [reply inc/umock_c/umock_c.h]: Correct, removed the WINDOWS part

**Pull request merged**